File: mergesort programs/mergesort.c
#include <stdio.h>
#include<stdlib.h>
void swap(int *x,int *y)
{
 int temp=*x;
 *x=*y;
 *y=temp;
}

void Merge(int A[],int l,int mid,int h)
{
 int i=l,j=mid+1,k=l;
 int B[100];

 while(i<=mid && j<=h)
 {
 if(A[i]<A[j])
 B[k++]=A[i++];
 else
 B[k++]=A[j++];
 }
 while(i<=mid)
 B[k++]=A[i++];
 while(j<=h)
 B[k++]=A[j++];

 for(i=l;i<=h;i++)
 A[i]=B[i];
}

void MergeSort(int A[],int l,int h)
{
 int mid;
 if(l<h)
 {
 mid=(l+h)/2;
 MergeSort(A,l,mid);
 MergeSort(A,mid+1,h);
 Merge(A,l,mid,h);

 }
}

int main()
{
 int A[]={1,3,2,7,4,8,0,5},n=8,i;

 MergeSort(A,0,n);

  printf("Sorted array is :- ");

 for(i=1;i<9;i++)
 printf("%d ",A[i]);
 printf("\n");

 return 0;
}

File: mergesort programs/mergesort.py
import numpy as np

def merge(left,mid,right,array) :
    x = z = left
    y = mid+1 
    sorted_array = np.array([None]*(right+1))
    while x <= mid and y <= right :
        if array[x] < array[y] :
            sorted_array[z] = array[x]
            z += 1
            x += 1
        else :
            sorted_array[z] = array[y]
            z += 1
            y += 1
    while x <= mid :
        sorted_array[z] = array[x]
        z += 1
        x += 1
    while y <= right :
        sorted_array[z] = array[y]
        z += 1
        y += 1
        
    x = left
    y = right
    while x <= y :
        array[x] = sorted_array[x]
        x += 1

def merge_sort(left,right,array) :
    if left < right :
        mid = (left + right)//2
        merge_sort(left,mid,array)
        merge_sort(mid+1,right,array)
        merge(left,mid,right,array)
        
array = [1,3,2,7,4,8,0,5]
merge_sort(0,len(array)-1,array)
print("sorted array is :- ", array)